bff-service/index.js

```diff
@@ -0,0 +1,86 @@
+const express = require('express');
+require('dotenv').config();
+const axios = require('axios').default;
+
+const app = express();
+const PORT = process.env.PORT || 3006;
+
+app.use(express.json());
+
+const key = '/products';
+const cache = new Map();
+
+const cashRequest = async (axiosConfig) => {
+  if (cache.has(key)) {
+    return cache.get(key);
+  } else {
+    await axios(axiosConfig)
+      .then((response) => {
+        cache.set(key, response.data);
+      });
+
+    const timer = setTimeout(() => {
```

Comment:
Better to create separated type for cache and move all logic related to revalidation and work with Map to this type.